src/Text/Gigaparsec/Internal/Token/Numeric.hs

```diff
@@ -371,23 +371,35 @@ mkSigned NumericDesc{..} !unsigned !err = IntegerParsers {
         _number = annotate (labelIntegerSignedNumber err Nothing) $
           atomic (sign <*> annotate (labelIntegerNumberEnd err) (number unsigned))
 
-{-type FloatingParsers :: *
-data FloatingParsers = FloatingParsers {}
+type FloatingParsers :: *
+data FloatingParsers = FloatingParsers { 
+  -- | Parse a single real number literal in decimal form (base 10).
+    decimalFloating :: Parsec Rational
+  -- | Parse a single real number literal in hexadecimal form (base 16).
+  , hexadecimalFloating :: Parsec Rational
+  -- | Parse a single real number literal in octal form (base 8).
+  , octalFloating :: Parsec Rational
+  -- | Parse a single real number literal in binary form (base 2).
+  , binaryFloating :: Parsec Rational
+  -- | Parse a single real number literal, 
+  -- which can be in many forms and bases depending on the configuration.
+  , numberFloating :: Parsec Rational
+  }
```

Comment:
While I can use `DuplicateRecordFields`, it's quite unwieldy when actually using the duplicated field selectors in the same module as it complains about ambiguous names. Therefore I added the `-Floating` suffix in the field names.

Alternatively, we could
- Define `FloatingParsers` in a new module
- Use `DuplicateRecordFields` and use `getField @"decimal" xxx` instead of `decimal xxx` to avoid the ambiguity. But `getField` is not Safe Haskell
- Maybe there are other ways that I'm not aware of

Reply:
There is a precedent set somewhere else for how to resolve the ambiguous names, I think? Just check around

Reply:
I'm not aware, please let me know 🙏🏼

Reply:
I think I ran into overloaded field names for integer parsers no? Maybe that was their base names themselves which was later changed in parsley too

Reply:
Looks like there's no `DuplicateRecordFields` extension used in the codebase.

Reply:
I mean I just added a suffix of some kind

Reply:
OK then, so all good here